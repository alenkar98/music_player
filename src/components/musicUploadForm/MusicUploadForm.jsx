import { useState } from 'react';
import './musicUploadForm.css';

export default function MusicUploadForm() {
  const [selectedFile, setSelectedFile] = useState(null);
  const [uploadProgress, setUploadProgress] = useState(0);
  const [isUploading, setIsUploading] = useState(false);

  const handleFileChange = (event) => {
    const file = event.target.files[0];
    setSelectedFile(file);
  };

  const handleSaveFile = () => {
    if (selectedFile) {
      setIsUploading(true);
      const interval = setInterval(() => {
        setUploadProgress((prevProgress) => {
          const newProgress = prevProgress + 10;
          if (newProgress >= 100) {
            console.log(selectedFile.name + ' uploaded successfully');
            clearInterval(interval);
            setIsUploading(false);
            setSelectedFile(null);
            setUploadProgress(0);
          }
          return newProgress;
        });
      }, 500);
    } else {
      console.log('No file selected');
    }
  };

  return (
    <div className={'upload'}>
      <input
        type="file"
        id="file-input"
        className={'input'}
        name="filename"
        accept={'.mp3,.wav'}
        onChange={handleFileChange}
      />
      <label className="file-input-label" htmlFor="file-input">
        {selectedFile ? selectedFile.name : 'Select a File'}
      </label>
      <label
        className={selectedFile ? 'file-input-label' : 'disabled'}
        onClick={selectedFile ? handleSaveFile : () => {}}
      >
        {isUploading ? `Uploading ${uploadProgress}%` : 'Upload'}
      </label>
    </div>
  );
}
