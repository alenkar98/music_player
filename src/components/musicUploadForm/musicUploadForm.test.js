import React from 'react';
import renderer from 'react-test-renderer';
import MusicUploadForm from './MusicUploadForm';

test('MusicUploadForm component renders correctly', () => {
  const component = renderer.create(<MusicUploadForm />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
