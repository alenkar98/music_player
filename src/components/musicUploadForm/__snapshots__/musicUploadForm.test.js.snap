// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MusicUploadForm component renders correctly 1`] = `
<div
  className="upload"
>
  <input
    accept=".mp3,.wav"
    className="input"
    id="file-input"
    name="filename"
    onChange={[Function]}
    type="file"
  />
  <label
    className="file-input-label"
    htmlFor="file-input"
  >
    Select a File
  </label>
  <label
    className="disabled"
    onClick={[Function]}
  >
    Upload
  </label>
</div>
`;
